### ISSUE TITLE
[Gui] Adventure-GUI doesn't load

### ISSUE THREAD

**Issue by vividos** [craftworkgames/MonoGame.Extended#552]
I'm trying to load the adventure-gui-skin.json Skin object using the current develop branch. Unfortunately some exceptions are thrown. I had to change the file in some parts. I'd like to review what has to be done to get it to work again. Ideally I could work on the issues in submitting PR's.

First I had to change the types; the classes have no Gui prefix anymore.
Next I had to change all the size properties, from e.g. "200 150" to { "Width": "200", "Height": "150" }. This would change how the size properties are loaded. Could there be a "custom importer" for some types, such as MonoGame.Extended.Size?

Then there are some controls missing, notably Image (found no source) and Dialog (found a Dialog.cs with all code commented out). How should this be fixed?

**Comment by craftworkgames** [comment]
> First I had to change the types; the classes have no Gui prefix anymore.

Yep, I decided to drop the `Gui` prefix because it was inconsistent with the naming conventions in the rest of the library. Everything is already in the `Gui` namespace anyway.

> Next I had to change all the size properties, from e.g. "200 150" to { "Width": "200", "Height": "150" }. This would change how the size properties are loaded. Could there be a "custom importer" for some types, such as MonoGame.Extended.Size?

This one surprised me a first, but I think maybe we've changed things from `Size2` to `Size`. The way to solve this properly is to implement a `JsonConverter`. There's some examples already in the `MonoGame.Extended.Serialization` namespace. Particularly, there's already a `Size2JsonConverter`. Once the converter is implemented it'll need to be added to the `GuiJsonSerializer`.

> Then there are some controls missing, notably Image (found no source) and Dialog (found a Dialog.cs with all code commented out). How should this be fixed?

This one is a bit more complicated. There's a much bigger job to be done to change how dialogs and other floating windows or popups (e.g. the combo box dropdown) are rendered. This stuff never worked particularly well in version 1 so my plan was to revamp all this for version 2.0

I've already got a few things on my plate right now so I'd be more than happy if you want to start doing some of this work. The `JsonConverter` stuff would be a good place to start.

Thanks mate.
